materials_io/dft.py

```diff
@@ -0,0 +1,95 @@
+from typing import Union, Iterable, Tuple
+from materials_io.base import BaseParser
+from dfttopif import files_to_pif
+from operator import itemgetter
+from glob import glob
+import itertools
+import os
+
+
+# List of files that are known to the VASP parser
+_vasp_file_names = ["outcar", "incar", "chgcar", "wavecar", "wavcar",
+                    "oszicar", "ibzcar", "kpoints", "doscar", "poscar",
+                    "contcar", "vasp_run.xml", "xdatcar"]
+
+
+class DFTParser(BaseParser):
+    """Extract data from Density Functional Theory calculation results
+
+    Uses the `dfttopif <https://github.com/CitrineInformatics/pif-dft>`_ parser to extract
+    metadata from each file
+    """
+
+    def __init__(self, quality_report=False):
+        """Initialize a featurizer
+
+        Args:
+            quality_report (bool): Whether to generate a quality report
+        """
+        self.quality_report = quality_report
+
+    def group(self, paths: Union[str, Iterable[str]], context: dict = None):
+        # Check arguments
+        if isinstance(paths, (str,)):
+            paths = [paths]
+
+        # Clean paths
+        paths = [os.path.abspath(os.path.expanduser(f)) for f in paths]
```

Comment:
Should this be a utility function? It's going to be very common to clean paths, but is also very short.

Reply:
I agree. I've broken it out into a new module